server/src/config.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) ZeroC, Inc.
+
+use crate::Backend;
+use tower_lsp::lsp_types::{ConfigurationItem, DidChangeConfigurationParams, Url};
+
+#[derive(Default, Debug)]
+pub struct SliceConfig {
+    pub reference_urls: Option<Vec<Url>>,
+}
+
+impl SliceConfig {
+    // Attempt to create a SliceConfig from the backend with the provided uri root.
+    pub async fn try_from_backend(
+        backend: &Backend,
+        root_uri: &Url,
+    ) -> tower_lsp::jsonrpc::Result<SliceConfig> {
+        let reference_directories = Self::fetch_reference_directories(backend).await?;
+        Self::create_config_from_directories(reference_directories, root_uri)
+    }
+
+    // Creates a SliceConfig from the provided configuration parameters and uri root.
+    pub fn from_configuration_parameters(
+        params: DidChangeConfigurationParams,
+        root_uri: &Url,
+    ) -> Self {
+        let reference_directories = Self::parse_reference_directories(&params);
+        Self::create_config_from_directories(reference_directories, root_uri).unwrap_or_default()
```

Comment:
So if there's an error parsing the reference directories we default to an empty config? Is this right?

Reply:
Correct, this keeps the SliceConfig as an accurate source of truth. When we need to access the directories to compile the slice we use `resolve_reference_paths` which then handles that if there are no referennce directories - to default to using the root_uri.

Reply:
`create_config_from_directories` returns a result, that can presumably fail for any number of reasons. Am I missing something?